lib/src/client/http2_channel.dart

```diff
@@ -31,27 +33,89 @@ class ClientChannel extends ClientChannelBase {
   final Object host;
   final int port;
   final ChannelOptions options;
+  final StreamController<ConnectionState> connectionStateStreamController =
+      StreamController.broadcast();
 
   ClientChannel(this.host,
       {this.port = 443, this.options = const ChannelOptions()})
       : super();
 
   @override
   ClientConnection createConnection() {
-    return Http2ClientConnection(host, port, options);
+    final connection = Http2ClientConnection(host, port, options);
+    connection.onStateChanged = (c) {
+      if (connectionStateStreamController.isClosed) {
+        return;
+      }
+      connectionStateStreamController.add(c.state);
+    };
+    return connection;
   }
+
+  @override
+  Future<void> shutdown() {
+    return super.shutdown()
```

Comment:
This code should probably use `whenComplete` instead of `then`, otherwise we will not close controller if `shutdown` throws an error. 

Alternatively you could write (which I personally consider easier to read): 

```dart
Future<void> shutdown() async {
  try {
    return super.shutdown();
  } finally {
    connectionStateStreamController.close();
  }
}
```

Reply:
Done in the new commit